build-scripts/lerna-publish.js

```diff
@@ -0,0 +1,144 @@
+#!/usr/bin/env node
+/*
+ * Copyright 2018 The Closure Compiler Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Custom lerna cli which adds a publication command to always attempt publication
+ * of all packages.
+ *
+ * Taken directly from the lerna publication command and lerna cli packages.
+ */
+
+'use strict';
+const cli = require('@lerna/cli');
+const addCmd = require('@lerna/add/command');
+const bootstrapCmd = require('@lerna/bootstrap/command');
+const changedCmd = require('@lerna/changed/command');
+const cleanCmd = require('@lerna/clean/command');
+const createCmd = require('@lerna/create/command');
+const diffCmd = require('@lerna/diff/command');
+const execCmd = require('@lerna/exec/command');
+const importCmd = require('@lerna/import/command');
+const initCmd = require('@lerna/init/command');
+const linkCmd = require('@lerna/link/command');
+const listCmd = require('@lerna/list/command');
+const publishCmd = require('@lerna/publish/command');
+const runCmd = require('@lerna/run/command');
+const versionCmd = require('@lerna/version/command');
+const Semver = require('semver');
+const pkg = require('lerna/package.json');
+const { PublishCommand } = require('@lerna/publish');
+const fs = require('fs');
+const path = require('path');
+
+function factory(argv) {
+  return new TravisPublishCommand(argv);
+}
+
+/** Override methods in the main publication command class to return the full set of packages for publication */
+class TravisPublishCommand extends PublishCommand {
+  findVersionedUpdates() {
```

Comment:
Could you add some documentation for this method to give enough context to understand this without having to hunt down the docs for PublishCommand?

Specifically:
1. what is this method supposed to return?
2. roughly what do detectFromGit() and detectCanaryVersions() do?